src/overlay121/DWWarp.c

```diff
@@ -0,0 +1,385 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "core_sys.h"
+
+#include "struct_decls/struct_02006C24_decl.h"
+#include "struct_decls/sys_task.h"
+#include "struct_decls/struct_020203AC_decl.h"
+
+#include "struct_defs/struct_020170F4.h"
+#include "struct_defs/struct_02017248.h"
+#include "struct_defs/struct_02017294.h"
+#include "struct_defs/struct_0207C690.h"
+#include "struct_defs/struct_02099F80.h"
+#include "overlay115/camera_angle.h"
+
+#include "unk_02002328.h"
+#include "unk_02005474.h"
+#include "overlay_manager.h"
+#include "narc.h"
+#include "unk_0200D9E8.h"
+#include "unk_0200F174.h"
+#include "unk_020170BC.h"
+#include "unk_02017728.h"
+#include "heap.h"
+#include "unk_0201E3D8.h"
+#include "gx_layers.h"
+#include "unk_02020020.h"
+#include "unk_0202419C.h"
+#include "unk_02024220.h"
+#include "overlay121/DWWarp.h"
+
+typedef struct {
+    GenericPointerData * p3DCallback;
+    UnkStruct_020203AC * camera;
+    SysTask * task;
+    int frame_cnt;
+    int soundEffect_cnt;
+    UnkStruct_02017294 animation_obj;
+    UnkStruct_02017248 animation_animation;
+    UnkStruct_02017248 animation_animation2;
+    UnkStruct_020170F4 animation_model;
+    u32 move_frame;
+    int camera_angle;
+    int camera_angle_calc;
+    fx32 camera_distance;
+    int camera_perspective;
+    NNSFndAllocator allocator;
+} DistortionWorld_Warp;
```

Comment:
**polish:** Use `camelCase` for struct-members to match style in the rest of the repo. Additionally, `typedef` structs with the same name as their declared name:

```
typdef struct DistortionWorldWarp {
    // members
} DistortionWorldWarp;
```